### PR TITLE
Feedback by Hannes

### PR DESCRIPTION
**Pull request by tlodderstedt**

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -169,25 +175,52 @@ This example shows a combined request asking for access to account information a
...
+This specification defines a set of common data fields that are designed to be usable across different types of APIs. These data fields MAY be combined in different ways depending on the needs of the API. All data fields are OPTIONAL for use by a given API definition. The allowable values of all fields are determined by the API being protected.
 
 `locations`:
 :   An array of strings representing the location of the resource or resource server. These strings are typically URIs identifying the
-    location of the RS.
+    location of the RS. The field `locations` allows a client to specify where it intends to use a certain authorization, i.e., it is  possible to unambiguously assign permissions to resource servers. In situations with multiple resource servers, this prevents unintended client authorizations (e.g. a `read` scope value potentially applicable for an email as well as a cloud service) through audience restriction.
```
Comment (jricher) [inline main.md:223]: I would rather see this as a security consideration than in the definition here, but keep a forward reference. So extract the explanation text and leave something like this:

```suggestion
    location of the RS. This field can allow a client to specify a particular RS, as discussed in {{security-considerations-location-field-values}}.
```

File: main.md [hunk 2]
```diff
@@ -424,14 +426,15 @@ In this example, the client wants to get access to account information and intia
...
 ]
 ```
+Figure: URL decoded authorization details.
 
 ## Relationship to "scope" parameter {#scope}
 
 `authorization_details` and `scope` can be used in the same authorization request for carrying independent authorization requirements. 
 
-The AS MUST consider both sets of requirements in combination with each other for the given authorization request. The details of how the AS combines these parameters are specific to the APIs being protected and outside the scope of this specification.
+Note: Combined use of `authorization_details` and `scope` is supported by this specification to allow existing OAuth-based applications to incrementally migration towards `authorization_details`. It is RECOMMENDED that a given API uses only one form of requirement specification.
```
Comment (jricher) [inline main.md:435]: Would suggest rewording this so it's not an aside:

```suggestion
Combined use of `authorization_details` and `scope` is supported by this specification in part to allow existing OAuth-based applications to incrementally migrate towards using `authorization_details` exclusively. It is RECOMMENDED that a given API use only one form of requirement specification. 
```

File: main.md [hunk 3]
```diff
@@ -447,8 +450,8 @@ This specification does not define extensions to the authorization response.
 
 # Authorization Error Response {#authz_details_error}
 
-The AS MUST refuse to process any unknown authorization data type or authorization details not conforming to the respective type definition. If any of the objects in `authorization_details` contains an unknown authorization data type or an object of known type but containing unknown elements or elements of the wrong type or elements 
-with invalid values or if required elements are missing, the AS MUST abort processing and respond with an error `invalid_authorization_details` to the client. 
+The AS MUST refuse to process any unknown authorization details type or authorization details not conforming to the respective type definition. If any of the objects in `authorization_details` contains an unknown authorization details type or an object of known type but containing unknown fields or fields of the wrong type or fields 
+with invalid values or if required fields are missing, the AS MUST abort processing and respond with an error `invalid_authorization_details` to the client.
```
Comment (jricher) [inline main.md:454]: It's more verbose but I think this should be spelled out:

```suggestion
The AS MUST refuse to process any unknown authorization details type or authorization details not conforming to the respective type definition. The AS MUST abort processing and respond with an error `invalid_authorization_details` to the client if any of the following are true of any of the objects in `authorization_details` structure:

- Contains an unknown authorization details type value,
- An object of known type but containing unknown fields,
- Contains fields of the wrong type for the authorization details type,
- Contains fields with invalid values for the authorization details type, or
- Missing required fields for the authorization details type.
```

File: main.md [hunk 4]
```diff
@@ -770,18 +780,19 @@ The following shows the contents of an example JWT for the payment initiation ex
    }
 }
 ```
+Figure: Example for authorization details in JWT-based access token.
 
 In this case, the AS added the following example claims:
```
Comment (jricher) [inline main.md:785]:
```suggestion
In this case, the AS added the following example claims to the JWT-based access token:
```

File: main.md [hunk 5]
```diff
@@ -770,18 +780,19 @@ The following shows the contents of an example JWT for the payment initiation ex
...
+* `debtorAccount`: API-specific field containing the debtor account. In the example, this account was not passed in the authorization details but selected by the user during the authorization process. The field `user_role` conveys the role the user has with respect to this particular account. In this case, they is the owner. This data is used for access control at the payment API (the RS).
 
 ## Token Introspection 
 
 In the case of opaque access tokens, the data provided to a certain RS is determined using the RS's identifier with the AS (see [@I-D.ietf-oauth-jwt-introspection-response], section 3). 
 
-The token introspection response provides the RS with the authorization details applicable to it as a top-level JSON element along with the claims the RS requires for request processing. 
+The token introspection response provides the RS with the authorization details applicable to it as a top-level JSON field along with the claims the RS requires for request processing.
```
Comment (jricher) [inline main.md:795]: I believe "element" is actually correct here since we're talking about JSON itself

```suggestion
The token introspection response provides the RS with the authorization details applicable to it as a top-level JSON element along with the claims the RS requires for request processing. 
```

File: main.md [hunk 6]
```diff
@@ -1106,9 +1138,15 @@ Specification Document(s):
 
 ## OpenID Connect {#openid}
 
+OpenID Connect [@OIDC] specifies the JSON-based `claims` request parameter that can be used to specify the claims a client (acting as OpenID Connect Relying Party) wishes to receive in a fine-grained and privacy-preserving way as well as assign those claims to certain delivery mechanisms, i.e. ID Token or userinfo response. 
+
+The combination of the scope value `openid` and the additional parameter `claims` can be used beside `authorization_details` in the same way as every non-OIDC scope value. 
+
+Alternatively, there could be an authorization details type for OpenID Connect. This section gives an example of what such an authorization details type could look like.
```
Comment (jricher) [inline main.md:1145]:
```suggestion
Alternatively, there could be an authorization details type for OpenID Connect. This section gives an example of what such an authorization details type could look like, but defining such a type is outside the scope of this specification.
```

File: main.md [hunk 7]
```diff
@@ -1335,10 +1380,14 @@ In this use case, the AS authenticates the requester, who is not the patient, an
...
 
+   -09
+
+   * Incorporated feedback by Hannes
+
    -08
 
    * formatting in authorization details type section 
-   * added example for `privileges` common data element 
+   * added example for `privileges` common data field
```
Comment (jricher) [inline main.md:1390]: I wouldn't change the changelog (this is a nit)

```suggestion
   * added example for `privileges` common data element
```

File: main.md [hunk 8]
```diff
@@ -1361,7 +1410,7 @@ In this use case, the AS authenticates the requester, who is not the patient, an
 
    -04 
 
-   * restructured draft for better readability
+   * restructured specification for better readability
```
Comment (jricher) [inline main.md:1413]: I wouldn't change the changelog (this is a nit)

```suggestion
   * restructured draft for better readability
```

**Review by jricher — CHANGES_REQUESTED** [review]
Overall reasonable changes, I have made a few suggestions to improve the affected text.

File: main.md [hunk 9]
```diff
@@ -1335,10 +1380,14 @@ In this use case, the AS authenticates the requester, who is not the patient, an
 
    [[ To be removed from the final specification ]]
 
+   -09
+
+   * Incorporated feedback by Hannes
```
Comment (bc-pi) [inline main.md:1385]:
```suggestion
   * Incorporated feedback by Hannes as document shepherd
```

File: main.md (hunk 1 above)
Comment (tlodderstedt) [reply main.md:223]: adopted your proposal
